test/e2e/main_test.go

```diff
@@ -130,7 +130,8 @@ func testMainWrapper(m *testing.M) int {
 	var registryProvisioned bool
 	catalogSourceName, catalogSourceNamespace, err = df.CreateCatalogSourceFromIndex(indexImage)
 	if err != nil {
-		df.Logger.Fatalf("Failed to create the CatalogSource custom resource using the %s index image: %v", indexImage, err)
+		df.Logger.Errorf("Failed to create the CatalogSource custom resource using the %s index image: %v", indexImage, err)
+		return 1
```

Comment:
We could add a helper function that logs a message to the error level and returns 1 every time we call it, but that might be getting too abstract for now.

Also, `logrus.Panicf` could also be utilized as panicking still allows [defer functions to be run](https://blog.golang.org/defer-panic-and-recover) but the stack trace is a bit difficult to parse in the e2e_test.log logs.